src/main/java/com/google/roomies/data/Listing.java

```diff
@@ -276,9 +294,29 @@ public Builder setListingPrice(String listingPrice) throws UnknownCurrencyExcept
         .setSharedPrice(listingData.get(SHARED_ROOM_PRICE).toString())
         .setSinglePrice(listingData.get(SINGLE_ROOM_PRICE).toString())
         .setListingPrice(listingData.get(LISTING_PRICE).toString())
+        .setComments(getAllCommentsFromCollection(document.getId()))
         .build());
     }
 
+  /**
+  * Gets all comments from a given listing.
+  * 
+  * @return list of comment instances
+  */
+  private static ImmutableList<Comment> getAllCommentsFromCollection(String listingId)
+      throws IOException, InterruptedException, ExecutionException {
+    List<QueryDocumentSnapshot> documents = 
+      DatabaseFactory.getDatabase().getAllCommentsInListing(listingId).get().getDocuments();
+
+    List<Comment> comments =
+      StreamSupport.stream(documents.spliterator(), /* parallel= */ false)
+      .map(document -> Comment.fromFirestore(document))
+      .flatMap(Optional::stream)
+      .collect(Collectors.toList());
```

Comment:
You can directly use toImmutableList method from com.google.common.collect.ImmutableList.

Reply:
Done